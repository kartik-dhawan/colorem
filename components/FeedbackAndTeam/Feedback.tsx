import { Box, Grid, Typography } from "@mui/material"
import { styles } from "./styles/styles"
import LikeCard from "./FeedbackCards/LikeCard"
import TextCard from "./FeedbackCards/TextCard"

const Feedback = () => {
  const fid = "feedbackSection"
  return (
    <Grid
      item
      md={5}
      xs={12}
      order={{ md: 1, xs: 2 }}
      className={fid + "Wrapper"}
      id={fid + "Wrapper"}
      sx={{
        borderRight: "1px solid #444",
      }}
    >
      <Typography
        variant="h5"
        className={fid + "Title"}
        id={fid + "Title"}
        sx={styles.feedbackSectionTitle}
      >
        Working on your feedbacks!
      </Typography>
      <Box className="feedbackCardsWrapper" id="feedbackCardsWrapper">
        <LikeCard likes={23} />
        <TextCard cardVariant={"feedback"} />
        <TextCard cardVariant={"comment"} />
      </Box>
    </Grid>
  )
}

export default Feedback
