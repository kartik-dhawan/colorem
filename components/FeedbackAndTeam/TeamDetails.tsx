import { Box, Button, Grid, Typography } from "@mui/material"
import Image from "next/image"
import Link from "next/link"
import { styles } from "./styles/styles"

const TeamDetails = () => {
  const tid = "teamSection"

  const avatarObj = [
    {
      name: "Kartik Dhawan",
      src: "/user1.JPG",
    },
    {
      name: "Myrtle Wilson",
      src: "/user2.PNG",
    },
    {
      name: "Jules Dawn",
      src: "/user3.JPG",
    },
  ]

  return (
    <Grid
      item
      md={7}
      xs={12}
      order={{ md: 2, xs: 1 }}
      className={tid + "Wrapper"}
      id={tid + "Wrapper"}
      sx={{}}
    >
      <Box
        className={tid + "Head"}
        id={tid + "Head"}
        sx={{
          display: "flex",
          alignItems: "center",
          margin: "0px 0px 25px 0px",
        }}
      >
        <Button
          variant="contained"
          className={tid + "Title"}
          id={tid + "Title"}
          sx={styles.teamSectionTitle}
        >
          <Link href="/">TEAM</Link>
        </Button>
        <Box
          className={tid + "AvatarWrapper"}
          id={tid + "AvatarWrapper"}
          sx={styles.teamSectionAvatarWrapper}
        >
          {avatarObj.map((avi, i: number) => {
            return (
              <Image
                key={i}
                src={`/images${avi.src}`}
                alt={avi.name + " Avatar"}
                height={60}
                width={60}
                className={tid + "Avatar"}
                id={tid + "Avatar_" + i}
              />
            )
          })}
        </Box>
      </Box>
      <Typography
        variant="body2"
        className={tid + "Body"}
        id={tid + "Body"}
        sx={styles.teamSectionBody} // ellipsis css in .scss file
      >
        The super fast color palettes generator! Create the perfect color
        palette or get inspired by thousands of beautiful color schemes. Explore
        trending palettes now & start creating.
      </Typography>
    </Grid>
  )
}

export default TeamDetails
