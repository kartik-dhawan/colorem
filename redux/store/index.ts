import { configureStore } from "@reduxjs/toolkit"
import countSlice from "../slices/countSlice"

const store = configureStore({
  reducer: {
    countSlice,
  },
})

export default store
