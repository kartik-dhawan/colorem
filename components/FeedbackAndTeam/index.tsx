import { Grid } from "@mui/material"
import Feedback from "./Feedback"
import TeamDetails from "./TeamDetails"

const FeedbackAndTeam = () => {
  return (
    <Grid container spacing={5} mt={3}>
      <Feedback />
      <TeamDetails />
    </Grid>
  )
}

export default FeedbackAndTeam
