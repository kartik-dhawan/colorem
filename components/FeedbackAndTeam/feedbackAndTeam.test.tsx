import { Provider } from "react-redux"
import renderer from "react-test-renderer"
import "@testing-library/jest-dom"
import store from "../../redux/store"
import { render } from "@testing-library/react"
import FeedbackAndTeam from "."
import Feedback from "./Feedback"
import TeamDetails from "./TeamDetails"

describe("Feedback and team section Component", () => {
  it("-- should match snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <FeedbackAndTeam />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("-- should render feedback component correctly", () => {
    const view = render(
      <Provider store={store}>
        <Feedback />
      </Provider>
    )
    const feedbackSection = view.getByTestId("feedbackSectionWrapper")
    expect(feedbackSection).toBeInTheDocument()
  })

  it("-- should render team details component correctly", () => {
    const view = render(
      <Provider store={store}>
        <TeamDetails />
      </Provider>
    )
    const teamSection = view.getByTestId("teamSectionWrapper")
    expect(teamSection).toBeInTheDocument()
  })
})
