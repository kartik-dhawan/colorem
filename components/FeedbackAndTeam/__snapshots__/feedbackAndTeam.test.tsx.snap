// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback and team section Component -- should match snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 css-1n8pt8h-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5 feedbackSectionWrapper css-14expfb-MuiGrid-root"
    data-testid="feedbackSectionWrapper"
    id="feedbackSectionWrapper"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 feedbackSectionTitle css-1pcczsu-MuiTypography-root"
      id="feedbackSectionTitle"
    />
    <div
      className="feedbackCardsWrapper MuiBox-root css-0"
      id="feedbackCardsWrapper"
    >
      <div
        className="feedbackCard MuiBox-root css-hkx9zj"
        id="feedbackCard"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium feedbackIcon css-i4bv87-MuiSvgIcon-root"
          data-testid="FavoriteIcon"
          focusable="false"
          id="feedbackIcon"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
          />
        </svg>
        <p
          className="MuiTypography-root MuiTypography-body1 css-rn997f-MuiTypography-root"
        >
          23
           new likes on palettes.
        </p>
      </div>
      <div
        className="feedbackCard MuiBox-root css-hkx9zj"
        id="feedbackCard"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium feedbackIcon css-i4bv87-MuiSvgIcon-root"
          data-testid="PersonOutlineIcon"
          focusable="false"
          id="feedbackIcon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
          />
        </svg>
        <p
          className="MuiTypography-root MuiTypography-body1 feedbackCardBody css-rn997f-MuiTypography-root"
        >
          Mr. Shay sent a
           
          <b>
            <i>
              feedback
            </i>
          </b>
          , 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores eaque dolor ad! ðŸ‡¦ðŸ‡· ðŸ’Ÿ'
        </p>
      </div>
      <div
        className="feedbackCard MuiBox-root css-hkx9zj"
        id="feedbackCard"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium feedbackIcon css-i4bv87-MuiSvgIcon-root"
          data-testid="CommentIcon"
          focusable="false"
          id="feedbackIcon"
          viewBox="0 0 24 24"
        >
          <path
            d="M21.99 4c0-1.1-.89-2-1.99-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4-.01-18zM18 14H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
          />
        </svg>
        <p
          className="MuiTypography-root MuiTypography-body1 feedbackCardBody css-rn997f-MuiTypography-root"
        >
          Mr. Shay sent a
           
          <b>
            <i>
              comment
            </i>
          </b>
          , 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores eaque dolor ad! ðŸ‡¦ðŸ‡· ðŸ’Ÿ'
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7 teamSectionWrapper css-16629bz-MuiGrid-root"
    data-testid="teamSectionWrapper"
    id="teamSectionWrapper"
  >
    <div
      className="teamSectionHead MuiBox-root css-w09u1u"
      id="teamSectionHead"
    >
      <a
        className="teamSectionTitle"
        href="/about/team"
        id="teamSectionTitle"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-fv8tu3-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          TEAM
        </button>
      </a>
      <div
        className="teamSectionAvatarWrapper MuiBox-root css-1ra01ct"
        id="teamSectionAvatarWrapper"
      />
    </div>
    <p
      className="MuiTypography-root MuiTypography-body2 teamSectionBody css-1tz1ww6-MuiTypography-root"
      id="teamSectionBody"
    />
  </div>
</div>
`;
